Enabling "glob" feature makes compilation fail on stable

```rust
error[E0658]: `#[doc(cfg)]` is experimental
   --> .../include_dir-0.7.1/src/lib.rs:102:1
    |
102 | #[doc(cfg(feature = "glob"))]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

I don't think you need it at all because since recently `#[cfg]` implies `#[doc(cfg)]` (https://github.com/rust-lang/rust/pull/89596).

Also, in some very rare cases when it may not work and you need to do it manually, you can do this:

- In `Cargo.toml`:
  ```toml
  [package.metadata.docs.rs]
  features = ["foo", "bar"]
  rustdoc-args = ["--cfg", "docsrs"]
  ```
- In `lib.rs`:
  ```rust
  #![cfg_attr(docsrs, feature(doc_cfg))]
  ```
- And then you can safely do:
  ```rust
  #[cfg_attr(docrs, doc(cfg(feature = "foo")))]
  ```
- And in order to build docs locally:
  ```sh
  RUSTDOCFLAGS="--cfg docsrs" cargo +nightly doc --features foo,bar
  ```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -25,18 +25,48 @@ jobs:
           profile: minimal
           toolchain: ${{ matrix.rust }}
           override: true
-      - uses: actions-rs/cargo@v1
+      - name: Check
+        uses: actions-rs/cargo@v1
         with:
           command: check
           args: --workspace --verbose
-      - uses: actions-rs/cargo@v1
+      - name: Build
+        uses: actions-rs/cargo@v1
         with:
           command: build
           args: --workspace --verbose
-      - uses: actions-rs/cargo@v1
+      - name: Test
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --workspace --verbose --no-default-features
+      - name: Test (default features)
+        uses: actions-rs/cargo@v1
         with:
           command: test
           args: --workspace --verbose
+      - name: Test (glob feature)
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --workspace --verbose --no-default-features --features glob
+      - name: Test (metadata feature)
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --workspace --verbose --no-default-features --features metadata
+      - name: Test (nightly feature)
+        uses: actions-rs/cargo@v1
+        if: matrix.rust == 'nightly'
+        with:
+          command: test
+          args: --workspace --verbose --no-default-features --features nightly
+      - name: Test (all features)
+        uses: actions-rs/cargo@v1
+        if: matrix.rust == 'nightly'
+        with:
+          command: test
+          args: --workspace --verbose --all-features
 
   api-docs:
     name: Publish API Docs to GitHub Pages

--- a/include_dir/src/file.rs
+++ b/include_dir/src/file.rs
@@ -1,6 +1,3 @@
-#[cfg(feature = "metadata")]
-use std::time::{Duration, SystemTime};
-
 use std::{
     fmt::{self, Debug, Formatter},
     path::Path,
@@ -40,6 +37,25 @@ impl<'a> File<'a> {
     /// The file's contents interpreted as a string.
     pub fn contents_utf8(&self) -> Option<&str> {
         std::str::from_utf8(self.contents()).ok()
+    }
+}
+
+#[cfg(feature = "metadata")]
+impl<'a> File<'a> {
+    /// Set the [`Metadata`] associated with a [`File`].
+    pub const fn with_metadata(self, metadata: crate::Metadata) -> Self {
+        let File { path, contents, .. } = self;
+
+        File {
+            path,
+            contents,
+            metadata: Some(metadata),
+        }
+    }
+
+    /// Get the [`File`]'s [`Metadata`], if available.
+    pub fn metadata(&self) -> Option<&crate::Metadata> {
+        self.metadata.as_ref()
     }
 }
 

--- a/include_dir/src/lib.rs
+++ b/include_dir/src/lib.rs
@@ -95,11 +95,9 @@ mod dir_entry;
 mod file;
 
 #[cfg(feature = "metadata")]
-#[doc(cfg(feature = "metadata"))]
 mod metadata;
 
 #[cfg(feature = "glob")]
-#[doc(cfg(feature = "glob"))]
 mod globs;
 
 #[cfg(feature = "metadata")]
@@ -108,7 +106,6 @@ pub use crate::metadata::Metadata;
 pub use crate::{dir::Dir, dir_entry::DirEntry, file::File};
 pub use include_dir_macros::include_dir;
 
-// #[cfg(rustdoc)]
 #[doc = include_str!("../README.md")]
 #[allow(dead_code)]
 fn check_readme_examples() {}

--- a/include_dir/src/metadata.rs
+++ b/include_dir/src/metadata.rs
@@ -1,28 +1,7 @@
 use std::time::{Duration, SystemTime};
 
-use crate::File;
-
-impl<'a> File<'a> {
-    /// Set the [`Metadata`] associated with a [`File`].
-    pub const fn with_metadata(self, metadata: Metadata) -> Self {
-        let File { path, contents, .. } = self;
-
-        File {
-            path,
-            contents,
-            metadata: Some(metadata),
-        }
-    }
-
-    /// Get the [`File`]'s [`Metadata`], if available.
-    pub fn metadata(&self) -> Option<&Metadata> {
-        self.metadata.as_ref()
-    }
-}
-
 /// Basic metadata for a file.
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
-#[doc(cfg(feature = "metadata"))]
 pub struct Metadata {
     accessed: Duration,
     created: Duration,

